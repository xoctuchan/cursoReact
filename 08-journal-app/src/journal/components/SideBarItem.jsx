import { Box, Divider, Drawer, Grid, List, ListItem, ListItemButton, ListItemIcon, ListItemText, Toolbar, Typography } from '@mui/material'
import { TurnedInNot } from '@mui/icons-material';
import React from 'react'
import { useMemo } from 'react';
import { setActiveNote } from '../../store/journal';
import { useDispatch } from 'react-redux';

export const SideBarItem = ({title, body, id, date, imageUrls}) => {

    const dispatch = useDispatch();

    const onClickSetActiveNote = ()=>{
        dispatch (setActiveNote({id, date, title, body, imageUrls}));
    }

    const newTitle = useMemo (()=>{
        return title.length >= 17 
            ? title.substring(0,17)+'...'
            : title ;
    },[title]);

    return (
        <ListItem disablePadding>
            <ListItemButton onClick={ onClickSetActiveNote }>
                <ListItemIcon>
                    <TurnedInNot />
                </ListItemIcon>
                <Grid container>
                    <ListItemText primary={ newTitle } />
                    <ListItemText secondary={ body } />
                </Grid>
            </ListItemButton>
        </ListItem>
    )
}
