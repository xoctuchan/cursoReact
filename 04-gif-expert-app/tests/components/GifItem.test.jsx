import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";


describe('Pruebas en GiftItem', () => { 
    const title= 'Prueba Title';
    const url = 'http://www.google.com/';
    test('Debe hacer match con el snapshot', () => { 
        const { container } = render (<GifItem title={title} url={url}/>);
        expect (container ).toMatchSnapshot();
    });

    test('Debe de mostrar la imagen con la url y el alt indicado', () => { 
        const { container } = render (<GifItem title={title} url={url}/>);
        const { src, alt }= screen.getByRole('img');
        expect( src ).toBe(url);
        expect( alt ).toBe(title);
        //console.log(screen.getByRole('img').src)
        //screen.debug();
    });

    test('Debe de mostrar el titulo en el componente', () => { 
        render (<GifItem title={title} url={url}/>);
        expect( screen.getByText(title) ).toBeTruthy();
    })
})