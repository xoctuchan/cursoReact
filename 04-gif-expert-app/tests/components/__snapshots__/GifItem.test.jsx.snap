// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GiftItem Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Prueba Title"
      src="http://www.google.com/"
    />
    <p>
      Prueba Title
    </p>
  </div>
</div>
`;
